torch/csrc/jit/pickler.cpp

```diff
@@ -86,13 +101,39 @@ void Pickler::addIValue(const IValue& ivalue) {
   } else if (ivalue.isString()) {
     pushMemoizedString(ivalue);
   } else if (ivalue.isGenericList()) {
-    pushList(ivalue);
+    pushGenericList(ivalue);
   } else if (ivalue.isGenericDict()) {
     pushDict(ivalue);
   } else if (ivalue.isNone()) {
     push<OpCode>(OpCode::NONE);
   } else if (ivalue.isIntList()) {
-    pushIntList(ivalue);
+    pushSpecializedList(
```

Comment:
Similarly, could template this to avoid copy pasta

Reply:
I am actually fine with this code because the template is actually going to be much hard to read (lots of templated methods on ivalue to achieve a very simple goal). If anything, a macro might be most appropriate.